woocommerce/woocommerce
# Split CALC_FOUND_ROW query into seperate count query for better performance

### All Submissions:






### Changes proposed in this Pull Request:



As described in the linked issue, this breaks up the CALC_FOUND_ROWS from the main query to convert it into a separate count query. See this [stack overflow link](https://stackoverflow.com/questions/186588/which-is-fastest-select-sql-calc-found-rows-from-table-or-select-count) I also verified the performance on a site with 8M orders and seems like the new approach is about at least 4x faster.

Closes #35464 






### How to test the changes in this Pull Request:



Since the issue is performance related, there are no functional changes. However, we should test if things are same or not:

With HPOS enabled, ensure the total order count is the same on the trunk branch and this branch. The count should not change because of this change (its just being calculated more efficiently in the background):

1. Main order listing page

<img width="1482" alt="Screenshot 2022-12-05 at 3 50 53 PM" src="https://user-images.githubusercontent.com/7571618/205613279-98ea24a2-f4a8-426b-bab5-b018da6a4fb5.png">

<img width="1322" alt="Screenshot 2022-12-05 at 3 51 45 PM" src="https://user-images.githubusercontent.com/7571618/205613464-fde5690c-402b-44fa-92e7-1df128a69baf.png">


3. A custom status page, such as "completed".

<img width="1469" alt="Screenshot 2022-12-05 at 3 53 26 PM" src="https://user-images.githubusercontent.com/7571618/205613998-e964d921-1366-444b-8480-4764fb4ec2ef.png">

<img width="1327" alt="Screenshot 2022-12-05 at 3 54 59 PM" src="https://user-images.githubusercontent.com/7571618/205614125-af13a7b6-1191-4417-b90a-0f4c5bf1dfaa.png">




### Other information:




### FOR PR REVIEWER ONLY:
